### DISCUSSION TITLE
Does adding a new Option field to a native_model struct need a separate version of the model?

### DISCUSSION THREAD

**Discussion by Gawdl3y** [vincent-herlemont/native_db#98] [Q&A] [answered]
In the case of adding a new field with an Option type to your existing native_model struct, does the model's version need to be incremented and an old version of the struct kept around to migrate from?
I'm assuming yes, but I want to verify!

**Comment by vincent-herlemont** [comment, marked as answer]
@Gawdl3y Yes, indeed, any change in a model must imperatively be managed with a new Rust type, which has the same id of `native_model` as the old one as well as a version increment. See the example code in the documentation of [`migrate`](https://docs.rs/native_db/latest/native_db/transaction/struct.RwTransaction.html#method.migrate ) or from the demo project [src-tauri/src/main.rs#L10-L43](https://github.com/vincent-herlemont/native_db_tauri_vanilla/blob/b9ecf3d7f086b87348c06708efbbbefefa80ead6/src-tauri/src/main.rs#L10-L43)
